frame/beefy/src/lib.rs

```diff
@@ -127,10 +131,14 @@ impl<T: Config> Pallet<T> {
 
 		assert!(<Authorities<T>>::get().is_empty(), "Authorities are already initialized!");
 
-		<Authorities<T>>::put(authorities);
+		let bounded_authorities =
+			BoundedSlice::<T::BeefyId, T::MaxAuthorities>::try_from(authorities)
+				.expect("Authorities vec too big");
```

Comment:
Is it ok to panic here?

Reply:
That's what we do in other pallets like authority discovery. I believe that it's OK since this function is only used in genesis build, and it basically panics during chain/runtime initialization, when it happens, people can just quickly shutdown their node and reconfigure it in their runtime config.

Reply:
Worth mentioning in a comment? Not trying to be nitpicky, more to get an intuition for what is obvious to a pallet-savvy reader and what isn't. :)

Reply:
I agree it is better to return a result here, unless you specify that this function should only be used in genesis, or in areas where panic is allowed.

Note that returning a result here should be easier to do now, since we have storage layer by default.